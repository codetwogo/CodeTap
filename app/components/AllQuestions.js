import React, { Component } from 'react';
import { Image, StyleSheet, TouchableOpacity, AsyncStorage, NetInfo } from 'react-native';
import { Container, Header, View, DeckSwiper, Card, CardItem, Thumbnail, Text, Left, Body, Icon, Button, Input } from 'native-base';

import HeaderComponent from './Header.js'

const cards = [
  {
    id: 'single-question-component',
    title: 'String Search (SAMPLE)',
    tests: [
      {
        inputs: ['or', 'hello world'],
        output: [true]
      }, {
        inputs: ['he', 'hello'],
        output: [true]
      }, {
        inputs: ['wet', 'youse sir'],
        output: [false]
      }
    ],
    boilerPlate: 'function indexOf (needle, haystack){\n\t\n}',
    description: 'You are attempting to find the index of the first appearance of one string (the needle) inside of another (the haystack).',
    image: require('./img/fullstack.png'),
    likes: 0
  }, {
    id: 'single-question-component',
    title: 'Reverse Array (SAMPLE)',
    tests: [
      {
        inputs: [[1, 2, 3, 4]],
        output: [[4, 3, 2, 1]]
      }
    ],
    boilerPlate: 'function reverseArray(arr){\n\t\n}',
    description: 'Write a function reverseArray that reverses the elements of an array and returns the reversed array.',
    image: require('./img/fullstack.png'),
    likes: 0
  },
];

export default class AllQuestions extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOnline: false,
      initializeCards: false,
      myQuestions: cards,
      onlineQuestions: [{}], // fetched from server
    }
    this.onQuestionPress = this.onQuestionPress.bind(this);

    //Sets the state based on connectivity
    //Initializes the myQuestions state
    const handleFirstConnectivityChange = (isConnected) => {
      console.log('Then, is ' + (isConnected ? 'online' : 'offline'));
      NetInfo.isConnected.removeEventListener(
        'change',
        handleFirstConnectivityChange
      );
      const result = this.state.isOnline === isConnected ? null : this.setState({ isOnline: isConnected });
      this.intializeComponent();
      return result;
    }
    NetInfo.isConnected.addEventListener(
      'change',
      handleFirstConnectivityChange
    );
  }

  //Populates state.myQuestions based connectivity
  intializeComponent() {
    if (this.state.isOnline) {
      fetch('https://c2g-question-maker.herokuapp.com/api/questions')
        .then(res => res.json())
        .then(resJson => {
          return AsyncStorage.setItem('myQuestions', JSON.stringify(resJson))
        })
        .then(() => {
          AsyncStorage.getItem('myQuestions')
            .then((value) => {
              var result = JSON.parse(value).reduce((prev, curr) => {
                return prev.concat(curr)
              }, []);
              this.setState(
                {
                  myQuestions: this.state.myQuestions.concat(result),
                  initializeCards: true
                })
            })
        })
    } else {
      //Used to clear local storage
      // AsyncStorage.setItem('myQuestions', JSON.stringify([]))

      AsyncStorage.getItem('myQuestions')
        .then((value) => {
          var result = JSON.parse(value).reduce((prev, curr) => {
            return prev.concat(curr)
          }, []);
          this.setState(
            {
              myQuestions: this.state.myQuestions.concat(result),
              initializeCards: true
            })
        })
    }
  }

  onQuestionPress(question) {
    this.props.navigator.push({ id: 'single-question-component', question: question })
  }
  render() {

    return (
      <Container style={styles.container}>
        <HeaderComponent navigator={this.props.navigator} style={styles.item} />

        <View style={{
          flex: 1
        }}>
        {/* DeckSwiper can only render dataSource once
        Boolean ensures that Questions are initialized beforehand */}
          {Boolean(this.state.initializeCards) && <DeckSwiper
            dataSource={this.state.myQuestions}
            renderItem={item => <TouchableOpacity onPress={() => {
              this.onQuestionPress(item)
            }}>
              <Card style={{
                backgroundColor: '#888888',
                elevation: 3
              }}>
                <CardItem style={{
                  backgroundColor: 'transparent'
                }}>
                  <Left>
                    <Thumbnail source={require('./img/fullstack.png')} />
                    <Body>
                      <Text>{item.title}</Text>
                      <Text not style={{
                        color: '#333'
                      }}>Medium</Text>
                    </Body>
                  </Left>
                </CardItem>
                <CardItem cardBody style={{
                  backgroundColor: '#aaaaaa',
                  padding: 10,
                  paddingTop: 10,
                  minHeight: 100,
                  alignItems: 'flex-start',
                  borderRadius: 10,
                  marginLeft: 10,
                  marginRight: 10
                }}>
                  <Text>{item.description}</Text>
                </CardItem>
                <CardItem style={{
                  backgroundColor: 'transparent'
                }}>
                  <Icon name="heart" style={{
                    color: '#ED4A6A'
                  }} />
                  <Text style={{
                    color: '#ccc'
                  }}>{item.likes}</Text>
                </CardItem>
              </Card>
            </TouchableOpacity>} />
          }
        </View>
        <View style={{
          flex: 1,
          justifyContent: 'flex-start',
          alignItems: 'center'
        }}>
          <Text style={{
            textAlign: 'center',
            color: '#888888',
            fontWeight: 'bold'
          }}>{`<-- Swipe Left and Right for Questions  -->`}</Text>
        </View>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#333333'
  },
  item: {}
});