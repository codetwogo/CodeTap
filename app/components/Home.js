import React, {Component} from 'react';
import {AppRegistry, View, Text, StyleSheet} from 'react-native';

export const HomeComponent = () => {
    return (
        <View style={styles.container}>
            <Text>Welcome to the Home Component</Text>
        </View>
    )
}

AppRegistry.registerComponent('HomeComponent', () => HomeComponent);

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#fff',
      alignItems: 'center',
      justifyContent: 'center',
    },
  });
  