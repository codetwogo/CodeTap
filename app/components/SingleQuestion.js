import React, { Component } from 'react';
import { AppRegistry, StyleSheet} from 'react-native';

import { Container, Content, Header, View, DeckSwiper, Card, CardItem, Thumbnail, Text, Left, Body, Icon, Button, Input } from 'native-base';

import HeaderComponent from './Header.js'

export default class SingleQuestionComponent extends Component {
    //consider destructuring {question: { title, tests, boilerPlate, description}}
    constructor(props) {
        super(props);
        this.state = {
            title: this.props.question.title,
            tests: this.props.question.tests,
            likes: this.props.question.likes,
            boilerPlate: this.props.question.boilerPlate,
            description: this.props.question.description
        };
        //consider Arrow functions
        this.onBackPress = this.onBackPress.bind(this);
        this.onCodePress = this.onCodePress.bind(this);
    }

    // go back to all questions list
    onBackPress() {
        this.props.navigator.push({
            id: 'all-questions-component'
        });
    }

    // navigate to the code environment to begin question
    onCodePress() {
        this.props.navigator.push({
            id: 'Code-Env',
            question: this.state
        });
    }

    render() {
        return (
          <Container>
            <HeaderComponent navigator={this.props.navigator} />
            <Content style={styles.container}>
              <View style={styles.topRowContainer}>
                <View style={{flexDirection: 'row', alignItems:'center'}}>
                  <Icon name="heart" style={{ fontSize: 24, color: '#ED4A6A' }} />
                  <Text style={{flex: 1, color: '#888888'}}>{` ${this.props.question.likes}`}</Text>
                  <Text style={{flex: 2, fontSize: 22, fontWeight: 'bold', color: '#aa0000', alignSelf: 'center'}}>{this.state.title}</Text>
                  <Text style={{flex: 1, color: '#880000', alignSelf: 'center'}}>{` `}</Text>
                </View>
              </View>
              <View style={{flex: 1, minHeight: 10, borderRadius: 10, padding: 10, paddingTop: 10, paddingBottom: 20, backgroundColor: '#777777', marginTop: 20, marginBottom: 16}}>
                {
                  this.state.tests.map(test => {
                    return (
                      <View key={test.inputs}>
                        <Text style={{lineHeight: 44}}>Inputs: {test.inputs}</Text>
                        <Text>Output: {test.output} </Text>
                      </View>
                    )
                  })
                }
              </View>
              <Text style={{padding: 5, fontSize: 16, marginBottom: 15, color: '#888888'}}>{this.state.description}</Text>
              <Button
                light
                onPress={this.onCodePress}
                bordered
                style={{borderColor: '#999999',alignSelf: 'center'}}>
                <Text style={{color: '#999999'}}>Code!</Text>
              </Button>
            </Content>
          </Container>
        );
    }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    paddingTop: 12,
    backgroundColor: '#333333'
  },
  topRowContainer: {

  }
});
