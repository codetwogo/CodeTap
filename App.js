import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Navigator } from 'react-native-deprecated-custom-components';

import HomeComponent from './app/components/Home';
import AllQuestions from './app/components/AllQuestions';
import SingleQuestion from './app/components/SingleQuestion';
import CodeEnv from './app/components/CodeEnv';
import TestEnv from './app/components/TestEnv';
// import BetaQuestions from './app/components/AllQuestionsBeta';

export default class App extends React.Component {

  renderScene(route, navigator) {

    switch (route.id) {

      case 'homecomponent':
        return (<HomeComponent navigator={navigator} />);

      case 'all-questions-component':
        return (<AllQuestions navigator={navigator} />);

      case 'single-question-component':
        return (<SingleQuestion navigator={navigator} question={route.question} />);

      case 'Code-Env' :
        return (<CodeEnv navigator={navigator} question={route.question}/>);

      case 'back-code-env':
        return (<CodeEnv navigator={navigator} userAnswer={route.userAnswer} textStates={route.textStates} question={route.question} />);

      case 'test-env':
        return (<TestEnv navigator={navigator} userAnswer={route.userAnswer} tests={route.tests} textStates={route.textStates} description={route.description}/>);

      case 'beta-questions':
        return (<BetaQuestions navigator={navigator} />);

      default:
        return (<HomeComponent navigator={navigator} />);
    }
  }

  render() {
    return (
      <Navigator
        initialRoute={{ id: 'homecomponent' }}
        renderScene={this.renderScene}
        configureScene={(route, routeStack) => Navigator.SceneConfigs.FloatFromLeft} />
    );
  }
}
