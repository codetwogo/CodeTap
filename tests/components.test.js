import React from 'react';
import App from '../App';
import Home from '../app/components/Home';
import AllQuestions from '../app/components/AllQuestions';
import SingleQuestion from '../app/components/SingleQuestion';

import renderer from 'react-test-renderer';
//import {shallow} from 'enzyme';
import shallowRenderer from 'react-test-renderer/shallow';
import { isDOMComponent, isComponentOfType, findWithType } from 'react-shallow-testutils';

const shallow = new shallowRenderer();

describe('The main App Entry Point...', () => {

    const component = renderer.create(<App />)

    it('renders without crashing', () => {
        const rendered = component.toJSON();
        expect(rendered).toBeTruthy();
    });

    it('matches snapshot', () => {
        const rendered = component.toJSON();
        expect(rendered).toMatchSnapshot();
        // console.log(rendered.find('Navigator'));
    })
})

describe('The Home Component...', () => {

    const component = renderer.create(<Home />)

    it('renders without crashing', () => {
        const rendered = component.toJSON();
        expect(rendered).toBeTruthy();
    });

    it('matches snapshot', () => {
        const rendered = component.toJSON();
        expect(rendered).toMatchSnapshot();
    })
})

describe('The AllQuestions Component...', () => {

    const component = renderer.create(<AllQuestions />)

    it('renders without crashing', () => {
        const rendered = component.toJSON();
        expect(rendered).toBeTruthy();
    });

    it('matches snapshot', () => {
        const rendered = component.toJSON();
        expect(rendered).toMatchSnapshot();
        // console.log(rendered.find('Navigator'));
    })
})

describe('The SingleQuestion Component...', () => {

    const component = renderer.create(<SingleQuestion />)

    it('renders without crashing', () => {
        const rendered = component.toJSON();
        expect(rendered).toBeTruthy();
    });

    it('matches snapshot', () => {
        const rendered = component.toJSON();
        expect(rendered).toMatchSnapshot();
        // console.log(rendered.find('Navigator'));
    })
})