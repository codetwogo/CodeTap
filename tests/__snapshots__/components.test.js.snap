// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AllQuestions Component... matches snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 30,
    }
  }
>
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 3,
      }
    }
    renderRow={[Function]}
    renderScrollComponent={[Function]}
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f4f4f4",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 3,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Question1
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f4f4f4",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 3,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Question2
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#f4f4f4",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 3,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Question3
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go back to base
      </Text>
    </View>
  </View>
</View>
`;

exports[`The Home Component... matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Welcome to the Home Component
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Start a War
      </Text>
    </View>
  </View>
</View>
`;

exports[`The SingleQuestion Component... matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    testing-question
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    tests
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    1234
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go Back
      </Text>
    </View>
  </View>
</View>
`;

exports[`The main App Entry Point... matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 1,
              },
              Object {
                "rotate": "0deg",
              },
              Object {
                "skewX": "0deg",
              },
              Object {
                "skewY": "0deg",
              },
            ],
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        >
          Welcome to the Home Component
        </Text>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#0C42FD",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Start a War
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
